types/wlr_output.c

```diff
@@ -247,3 +250,20 @@ uint16_t wlr_output_get_gamma_size(struct wlr_output *output) {
 	}
 	return output->impl->get_gamma_size(output);
 }
+
+static void rgba_to_argb(uint32_t *data, size_t height, size_t stride) {
+	size_t n = height*stride/4;
+	for (size_t i = 0; i < n; ++i) {
+		uint32_t v = data[i];
+		uint32_t rgb = (v & 0xffffff00) >> 8;
+		uint32_t a = v & 0x000000ff;
+		data[i] = rgb | (a << 24);
+	}
+}
+
+void wlr_output_read_pixels(struct wlr_output *output, void *out_data) {
+	wlr_output_make_current(output);
+	glReadPixels(0, 0, output->width, output->height, GL_RGBA, GL_UNSIGNED_BYTE,
```

Comment:
This should go through the renderer instead of putting GL calls in wlr_output. And do the rgba->argb translation in there too, the pixel format is specific to the renderer so it needs to be converted to the standard format by the renderer.